transit_odp/pipelines/tests/test_dataset_etl/test_servicepattern.py

```diff
@@ -97,3 +102,87 @@ def test_servicepattern(self):
                 break
 
         self.assertTrue(found)
+
+
+columns_lines = [
+    "line_id",
+    "line_name",
+    "outbound_description",
+    "inbound_description",
+]
+
+columns_service_patterns = [
+    "service_code",
+    "direction",
+    "geometry",
+    "localities",
+    "admin_area_codes",
+    "description",
+    "line_name",
+]
+
+
+@override_flag("is_timetable_visualiser_active", active=True)
+class ETLServicePatterns(ExtractBaseTestCase):
+    test_file = "data/test_servicepattern/test_etl_service_pattern.xml"
+
+    def test_extract_lines(self):
+        extracted = self.xml_file_parser._extract(self.doc, self.file_obj)
+
+        self.assertEqual(extracted.operating_profiles.shape[0], 20)
+
+        self.assertCountEqual(
+            list(extracted.lines.columns),
+            columns_lines,
+        )
+
+        self.assertEqual(extracted.operating_profiles.index.names, ["file_id"])
```

Comment:
Can we add count for extract step too?

Reply:
service pattern is not extracted. It's formed in the transformation step.

Reply:
we can add for lines only